### PR TITLE
mention dragonfly in readme and add multi queue benchmark info

### PR DESCRIPTION
**Pull request by kostasrim**
* mention dragonfly as an alternative to redis
* add multi queue benchmark results with dragonfly

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -56,6 +56,26 @@ Most of Sidekiq's overhead is Redis network I/O.
 ActiveJob adds a notable amount of CPU overhead due to argument deserialization and callbacks.
 Concurrency of 30 was determined experimentally to maximize one CPU without saturating it.
 
+Performance Dragonfly
```
Comment (kostasrim) [inline README.md:59]: @mperham I don't want to get intrusive so feel free to suggest changes or move this around. I also did not provide a comparison with Redis which I can happily add here (we will publish some content related to the comparison)

File: README.md [hunk 2]
```diff
@@ -14,7 +14,7 @@ Rails to make background processing dead simple.
 Requirements
 -----------------
 
-- Redis: 6.2+
+- Redis: 6.2+ or Dragonfly
```
Comment (mperham) [inline README.md:17]: Let's be explicit with a minimum version. 1.13+?

File: README.md (hunk 1 above)
Comment (mperham) [reply README.md:59]: I don't think this content is relevant to 99% of Sidekiq users so I'd prefer to move it to a Dragonfly-specific wiki page, which can be linked from the README.

File: README.md [hunk 3]
```diff
@@ -56,6 +56,26 @@ Most of Sidekiq's overhead is Redis network I/O.
...
+Performance Dragonfly
+-------------------
+
+The benchmark above was slightly revamped in `bin/multi_queue_bench` to scale better with multi-core systems like Dragonfly.
+We used a c5.4xlarge to run Dragonfly and a monstrous 96 core c5a.24xlarge to run the benchmark. All job types are no-op
+Sidekiq::Job. We run Dragonfly with `dragonfly --proactor_threads=n --shard_round_robin_prefix="queue"`. `n` is the number 
+of threads that Dragonfly uses and it defaults to the number of threads available on the system. This is different than Redis, 
+which is single threaded and scales vertically via the Redis cluster. This benchmark is all about scale, and therefore we
+run it with 3 different setups. `RUBY_YJIT_ENABLE=1 PROCESSES=96 QUEUES=k THREADS=10 ./multi_queue_bench` with `k` 1, 2, 8.
+We used Ruby with version 3.0.2p107 and Sidekiq trunk.
```
Comment (mperham) [inline README.md:68]: Ruby 3.0 is quite old and almost EOL. I would urge you to redo the benchmarks with Ruby 3.3.

Comment (kostasrim) [reply README.md:68]: I rerun the benchmarks with 3.3.0, roughly same performance :)

File: README.md (hunk 1 above)
Comment (kostasrim) [reply README.md:59]: I am not familiar on how to edit a `wiki page`. Is this something I can open a PR or?

Comment (mperham) [reply README.md:59]: You’d need to provide the content (the Using Redis page is a good reference or starting point). I’ll create one on Monday.

File: README.md (hunk 3 above)
Comment (mperham) [reply README.md:68]: Then please update the text.

**Comment by Niennienzz** [comment]
Hi @mperham , thank you so much for the feedback!
My name is Joe, and I am a developer advocate at DragonflyDB, very nice to e-meet you.
Kostas is away this week, and I will be helping with the contribution to the Wiki pages.
Please let me know the way to do so. Should I provide a Markdown file to you, or should I follow instructions like [this](https://gist.github.com/omaraboumrad/35654da0a376c57a2e0ab4d92ad0c339)?

Thanks.

**Comment by mperham** [comment]
Hey Joe, I started a gist with the basic Redis content we have today. You are welcome to fork it, wordsmith it as necessary and I will publish the new wiki page once we are both happy.

https://gist.github.com/mperham/949f86dc526c2eb5dc766d93a45c1267

**Comment by Niennienzz** [comment]
> Hey Joe, I started a gist with the basic Redis content we have today. You are welcome to fork it, wordsmith it as necessary and I will publish the new wiki page once we are both happy.
> 
> https://gist.github.com/mperham/949f86dc526c2eb5dc766d93a45c1267

Thank you so much! Working on it now.

**Comment by mperham** [comment]
Done! https://github.com/sidekiq/sidekiq/wiki/Using-Dragonfly

**Comment by Niennienzz** [comment]
> Done! https://github.com/sidekiq/sidekiq/wiki/Using-Dragonfly

Thank you so much! Although we still would like to add the little touch of `Redis: 6.2+ (or Dragonfly 1.13+)` on the homepage README, I will make another PR instead. Again, thanks for the help with the Wiki page. 😃

**Pull request closed without merging**